internal/worker/firewaller/firewaller.go

```diff
@@ -1095,6 +1105,16 @@ func (fw *Firewaller) flushModel() error {
 	if fw.environModelFirewaller == nil {
 		return nil
 	}
+
+	// We may be in a situation where we have added a model but doesn't have machines yet.
+	// To prevent the firewall worker from infinitely polling the Neutron security group API
+	// we skip flushing a model if we have no machines.
```

Comment:
This is leaking provider specific details. Just say something along the lines of we don't want to create the model specific artefacts until there's at least one machine in the model.